### PR TITLE
Issue #523 added README.md with updated python code in dynamic programming rod cutting #gssoc19

### PR DESCRIPTION
**Pull request by dshahid380**
This PR is about adding `README.md` 
Prefer Issue - #523  

##### Before
 - No README.md was there in Dynamic_Programming_Rod_Cutting folder.
 

##### After
 - README.md is added with full documentation in Dynamic_Programming_Rod_Cutting folder

### CODE REVIEW

File: Dynamic_Programming_Rod_Cutting/README.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+## Dynamic Programming - Rod Cutting
+We are given a rod of length n and we want to sell it. In order to sell it, we can either sell it as is or cut it into 
+pieces and sell them separately. The selling price is directly dependent on the length of the piece. <br><br>
+The only restriction is that we can only cut the rod into pieces of integer length. <br><br>
+We are also given an array of prices, also of length n. This array contains the price for selling a rod at a certain length.
+For example, prices[5] shows the price we can sell a rod of length 5. Generalising, prices[x] shows the price a rod of 
+length x can be sold.We are tasked to find the optimal solution to sell the given rod and prices array.
+### Approach
```
Comment (jainaman224) [inline Dynamic_Programming_Rod_Cutting/README.md]: Add new line above this.

File: Dynamic_Programming_Rod_Cutting/README.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+The function representing our approach is the following:<br>
+  
+  **Profit(n) = Max{ Price(n), Price(i) + Profit(n-i) }** <br> <br>
+Now we need to find our base case. For that, we will look at our restriction. We can only cut our rod in integer bits.
+So the smallest length we can get is 1. One could say that the smallest length of a rod is 0, but that case is not possible
+as rod of length 0 means that we don’t have a rod.<br><br>
+So, if we get a rod of length 1, we can only sell it as is. The profit of this sale is prices[1].<br><br>
+We have developed the method we will follow and we have found our base case, so we can head right into the code.
+### Implementation
```
Comment (jainaman224) [inline Dynamic_Programming_Rod_Cutting/README.md]: Add new line above this.

File: Dynamic_Programming_Rod_Cutting/README.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+The values of this array will be initialized at -1 and we will update them every time we find a better solution.
+We chose -1 so that we can easily check if we have found a solution for the given length.<br><br>
+Moreover, we will need our array of prices. The first value (for index=0) should be 0, as we cannot sell for 
+profit a rod of no length. You can set the rest to any value you want, in any order. Ascending order makes 
+intuitive sense; as the length increases so does the price. You don’t have to follow intuition though; 
+be a rebel! (although another order might produce weird or trivial results)
+<br><br>
+Lastly, we need to print the solution. For that, we call the function SellRod(n).
+
+```
```
Comment (jainaman224) [inline Dynamic_Programming_Rod_Cutting/README.md]: Remove this code.

**Comment by dshahid380** [comment]
@jainaman224  sir should i delete python code file?

**Comment by jainaman224** [comment]
@dshahid380 Yes please remove python code file.

**Comment by dshahid380** [comment]
@jainaman224  sir I have removed the python file and python code from the README.md as well.

**Comment by jainaman224** [comment]
No, you are doing it wrong.
You have removed it from main repo. You just have remove changes made by you. If you facing issue. You can ask us on GSSoC slack grouo.

**Comment by dshahid380** [comment]
@jainaman224 sir I have again added the same file file which I have removed.

**Comment by jainaman224** [comment]
@dshahid380 Can we move to slack public channel. Basically you are making changes to an existing file which do not add any value to it.

**Comment by jainaman224** [comment]
@dshahid380 Are you still working on it?

**Comment by dshahid380** [comment]
@jainaman224 , i am closing this PR and creating a new because there is too many changes are there which i dont want and not able to fix it.

**Pull request closed without merging**